### ISSUE TITLE
Lint suggestion: use `Path::is_empty()` instead of manual comparisons

### ISSUE THREAD

**Issue by DanielEScherzer** [rust-lang/rust-clippy#17615]
### What it does

Warn when comparing with `Path::new("")` and suggest using `path.is_empty()`

### Advantage

- More performant

### Drawbacks

- Churn

### Example

https://github.com/uutils/coreutils/blob/38ffca87fbd8e30fbfde57591f37ed77673874b6/src/uu/mkdir/src/mkdir.rs#L219 (assuming that uutils started requiring 1.98+)
```rust
    if path == Path::new("") {
        return Ok(());
    }
```

Could be written as:

```rust
    if path.is_empty() {
        return Ok(());
    }
```

### Comparison with existing lints

Unlike `comparison_to_empty` this is a comparison to another object rather than to an empty string or slice, so suggesting a new lint rather than an expansion to that

### Additional Context

Grep shows a number of hits in public code bases, and there are probably more that aren't caught by this regex due to using more qualified paths: https://grep.app/search?f.lang=Rust&regexp=true&q=%5B%21%3D%5D%3D%5Cs%2BPath%3A%3Anew%5C%28%22%22%5C%29

rust-lang/rust#148494
